### ISSUE TITLE
Shared ScreenModel between screens

### ISSUE THREAD

**Issue by renaudmathieu** [adrielcafe/voyager#168]
> Voyager by default expect that it screens can be stored inside a [Bundle](https://developer.android.com/guide/components/activities/parcelables-and-bundles). This means both Java Serializable and Parcelable are supported. By default all Voyager Screen is Java Serializable this means that Screen can be restored if all parameters are Java Serializable.

Is there a way to share the same ScreenModel between screens?

**Comment by programadorthi** [comment]
Well, a ScreenModel is remembered by Compose SlotTable and associated to a Screen. The idea is the same as Android ViewModel. Can you share a ViewModel with to Activities?
So, a hack try to you have a ScreenModel available to two Screen is:

1. Create a Singleton Screen.
2. In both Screen do: `val model by SingletonScreen.rememberScreenModel<YourScreenModel>()`

I have not tested that but as you can see [in the code](https://github.com/adrielcafe/voyager/blob/27920583c38430e98a2d85e7736d60ff1431138d/voyager-core/src/commonMain/kotlin/cafe/adriel/voyager/core/model/ScreenModel.kt#L29) a remembered ScreenModel is associated to a Screen. So, a Singleton Screen should works

**Comment by osrl** [comment]
but you can share a viewmodel with two fragments via navhost. it's a pretty common usage.

**Comment by renaudmathieu** [comment]
I've found a temporary solution using Koin.

1. Add the following dependencies:
```kotlin
implementation("io.insert-koin:koin-core:3.4.3")
implementation("io.insert-koin:koin-compose:1.0.4")
```

2. Start Koin and define a Single ScreenModel
```kotlin
KoinApplication(application = {
        modules(appModule)
}) {
```

3. Then you can inject it
```kotlin
val appScreenModel = koinInject<AppScreenModel>()
```

**Comment by osrl** [comment]
but this is scoped to entire app. it's never disposed right?

**Comment by Ric-AB** [comment]
Has anyone found a solution to this?

**Comment by osrl** [comment]
checkout #217

**Comment by DevSrSouza** [comment]
I think the main question here is: Where do we will scope it.

If you are running Single Activity, and using only one Navigator for the hole app, there is no place to Scope and free de ScreenModel/ViewModel, we could have a be scoped at the Navigator level but it will always stay alive.
If you are running Multiple Activities, one per flow, you can use for example [Kodein Scoped Injection](https://kosi-libs.org/kodein/7.19/framework/android.html#_lifecycle_scope) or [Koin](https://insert-koin.io/docs/reference/koin-android/scope/#android-scope-api).
If you are running Single Activity but Nested Navigation for scoping features, in this case, the #217 should be what you are looking for.

**Comment by DevSrSouza** [comment]
I had add new API for Navigator scoped ScreenModel, checkout, it is available starting from `1.0.0-rc08`
https://github.com/adrielcafe/voyager/pull/233

Docs: https://voyager.adriel.cafe/screenmodel#navigator-scoped-screenmodel
